parsec/core/fs/workspacefs/workspacefile.py

```diff
@@ -47,19 +49,22 @@ class WorkspaceFile:
     mode("r") -> read file.
     """
 
-    def __init__(self, fd: int, transactions, path: FsPath, mode: str = "r"):
-        self._fd = fd
+    def __init__(self, workspace, path, mode: str = "r"):
+        self._workspace = workspace
+        self._fd = None
         self._offset = 0
         self._state = FileState.INIT
-        self._path = path
-        self._transactions = transactions
+        self._path = FsPath(path)
+        self._transactions = workspace.transactions
         mode = mode.lower()
         # Preventing to open in write and read in same time or write and append or open with no mode
-        if sum(c in mode for c in "rwa") != 1:
+        if sum(c in mode for c in "rwax") != 1:
             raise ValueError("must have exactly one of create/read/write/append mode")
         # Preventing to open with non-existant mode
-        elif re.search("[^arwb]", mode) is not None:
-            raise ValueError("invalid mode: ", mode)
+        elif re.search("[^arwxb+]", mode) is not None:
+            raise ValueError(f"invalid mode: '{mode}'")
+        if "b" not in mode:
+            raise NotImplementedError("Must use binary mode")
```

Comment:
Maybe `"Text mode is not supported at the moment"`?

Reply:
Done 👍